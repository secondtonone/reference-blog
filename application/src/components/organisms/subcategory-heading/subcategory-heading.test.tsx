/**
 * @jest-environment jsdom
 */
import { render } from '@testing-library/react';

import fixtureSubcategoryGroupsNoPagedProps from '~/__fixtures__/subcategory-groups-no-paged';

import SubcategoryHeading, { ISubcategoryHeadingProps } from '.';

const subcategoryHeadingProps: ISubcategoryHeadingProps = (
  fixtureSubcategoryGroupsNoPagedProps as unknown);

describe('Match snapshots SubcategoryHeading', () => {
  let props: ISubcategoryHeadingProps;

  beforeEach(() => {
    props = { ...subcategoryHeadingProps };
  });

  it('default', () => {
    const { asFragment } = render(
      <SubcategoryHeading
        {...props}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('without name', () => {
    delete props.name;
    const { asFragment } = render(
      <SubcategoryHeading
        {...props}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('without description', () => {
    delete props.description;
    const { asFragment } = render(
      <SubcategoryHeading
        {...props}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('without previewImage', () => {
    delete props.previewImage;
    const { asFragment } = render(
      <SubcategoryHeading
        {...props}
      />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
