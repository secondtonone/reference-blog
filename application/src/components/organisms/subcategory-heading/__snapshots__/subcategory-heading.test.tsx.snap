// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots SubcategoryHeading default 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 subcategory-heading__SubcategoryHeadingStyled-sc-1cceahb-0 grLIrD"
    data-test="subcategory-heading"
  >
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 iymPtp"
      >
        <h1
          class="text-content__TextStyled-sc-1stkrsz-0 cpsJko subcategory-heading__TitleStyled-sc-1cceahb-2 CBOnN"
          data-test="subcategory-heading-name"
        >
          What Is SEO
        </h1>
        <span
          class="text-content__TextStyled-sc-1stkrsz-0 dbRYaR subcategory-heading__TitleStyled-sc-1cceahb-2 CBOnN"
          data-test="subcategory-heading-description"
        >
          Search Engine Optimization is the backbone of any digital marketing strategy. On this page you’ll find top SEO articles built for industry experts by industry experts. Covering all the hot SEO topics of the moment, explore our news, views and in-depth studies.
        </span>
      </div>
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 lnbOlo subcategory-heading__ImageStyled-sc-1cceahb-1 klhEIg"
        data-test="subcategory-heading-image"
      >
        <img
          alt="What Is SEO"
          class="image__ImageStyled-sc-rj8okk-0 hueYwk lazyload"
          data-src="https://static.example.com/blog/uploads/media/d6/6c/d66cd227c24cf1fb9ca43417e3cfda7f/improve-seo.svg"
          data-test="image"
          height="240"
          width="277"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Match snapshots SubcategoryHeading without description 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 subcategory-heading__SubcategoryHeadingStyled-sc-1cceahb-0 grLIrD"
    data-test="subcategory-heading"
  >
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 iymPtp"
      >
        <h1
          class="text-content__TextStyled-sc-1stkrsz-0 cpsJko subcategory-heading__TitleStyled-sc-1cceahb-2 CBOnN"
          data-test="subcategory-heading-name"
        >
          What Is SEO
        </h1>
      </div>
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 lnbOlo subcategory-heading__ImageStyled-sc-1cceahb-1 klhEIg"
        data-test="subcategory-heading-image"
      >
        <img
          alt="What Is SEO"
          class="image__ImageStyled-sc-rj8okk-0 hueYwk lazyload"
          data-src="https://static.example.com/blog/uploads/media/d6/6c/d66cd227c24cf1fb9ca43417e3cfda7f/improve-seo.svg"
          data-test="image"
          height="240"
          width="277"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Match snapshots SubcategoryHeading without name 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 subcategory-heading__SubcategoryHeadingStyled-sc-1cceahb-0 grLIrD"
    data-test="subcategory-heading"
  >
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 iymPtp"
      >
        <span
          class="text-content__TextStyled-sc-1stkrsz-0 dbRYaR subcategory-heading__TitleStyled-sc-1cceahb-2 CBOnN"
          data-test="subcategory-heading-description"
        >
          Search Engine Optimization is the backbone of any digital marketing strategy. On this page you’ll find top SEO articles built for industry experts by industry experts. Covering all the hot SEO topics of the moment, explore our news, views and in-depth studies.
        </span>
      </div>
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 lnbOlo subcategory-heading__ImageStyled-sc-1cceahb-1 klhEIg"
        data-test="subcategory-heading-image"
      >
        <img
          class="image__ImageStyled-sc-rj8okk-0 hueYwk lazyload"
          data-src="https://static.example.com/blog/uploads/media/d6/6c/d66cd227c24cf1fb9ca43417e3cfda7f/improve-seo.svg"
          data-test="image"
          height="240"
          width="277"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Match snapshots SubcategoryHeading without previewImage 1`] = `
<DocumentFragment>
  <div
    class="box-adaptive__BoxStyled-sc-g2wy39-0 subcategory-heading__SubcategoryHeadingStyled-sc-1cceahb-0 grLIrD"
    data-test="subcategory-heading"
  >
    <div
      class="box-adaptive__BoxStyled-sc-g2wy39-0 eZa-dHq"
    >
      <div
        class="box-adaptive__BoxStyled-sc-g2wy39-0 iymPtp"
      >
        <h1
          class="text-content__TextStyled-sc-1stkrsz-0 cpsJko subcategory-heading__TitleStyled-sc-1cceahb-2 CBOnN"
          data-test="subcategory-heading-name"
        >
          What Is SEO
        </h1>
        <span
          class="text-content__TextStyled-sc-1stkrsz-0 dbRYaR subcategory-heading__TitleStyled-sc-1cceahb-2 CBOnN"
          data-test="subcategory-heading-description"
        >
          Search Engine Optimization is the backbone of any digital marketing strategy. On this page you’ll find top SEO articles built for industry experts by industry experts. Covering all the hot SEO topics of the moment, explore our news, views and in-depth studies.
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
